cobra/flux_analysis/fastcc.py

```diff
@@ -59,28 +100,26 @@ def fastcc(model, flux_threshold=1.0, zero_cutoff=None):
     """
     zero_cutoff = normalize_cutoff(model, zero_cutoff)
 
+    rxns_to_remove = []
+
     with model:
-        obj_vars = []
-        vars_and_cons = []
-        prob = model.problem
-
-        for rxn in model.reactions:
-            var = prob.Variable("auxiliary_{}".format(rxn.id),
-                                lb=0.0, ub=flux_threshold)
-            const = prob.Constraint(rxn.forward_variable +
-                                    rxn.reverse_variable -
-                                    var, name="constraint_{}".format(rxn.id),
-                                    lb=0.0)
-            vars_and_cons.extend([var, const])
-            obj_vars.append(var)
-        model.add_cons_vars(vars_and_cons)
-
-        model.objective = prob.Objective(Zero, sloppy=True, direction="max")
-        model.objective.set_linear_coefficients({v: 1.0 for v in obj_vars})
-
-        sol = model.optimize()
-
-    rxns_to_remove = sol.fluxes[sol.fluxes.abs() < zero_cutoff].index
+        add_fastcc_cons_and_vars(model, flux_threshold)
+
+        # TODO: randomly set to 3; should decide on a default
+        for _ in range(3):
```

Comment:
This should run until no more reactions can be made to carry flux. Also this is missing the main part of FastCC which is setting objective coefficients for only those reactions that have not yet been marked as non-blocked. So basically you would need an additional function that updates the objective and takes a list of reactions to either include or exclude in the objective. Then pass in the the set of reactions that have not been flagged yet in each iteration.

Reply:
Ah, I see the point I have been missing. I'll give the paper another read.